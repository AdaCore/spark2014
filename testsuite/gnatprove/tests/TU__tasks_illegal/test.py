from test_support import prove_all

prove_all(opt=["-P", "test1.gpr", "-u", "t.adb"])
prove_all(opt=["-P", "test2.gpr", "-u", "t.adb"])
prove_all(opt=["-P", "test3.gpr", "-u", "t.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "po_t.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "po_t2.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "po_t3.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "po_t4.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "po_t5.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "main.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "po_t7.adb"])
prove_all(opt=["-P", "test.gpr", "-u", "use_po_t8.adb"])
