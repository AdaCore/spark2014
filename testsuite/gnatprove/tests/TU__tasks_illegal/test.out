t.adb:26:07: sequential elaboration supported only in restricted profile
t.adb:27:07: sequential elaboration supported only in restricted profile
t.adb:28:07: sequential elaboration supported only in restricted profile
gnatprove: error during generation of Global contracts
t.adb:26:07: from profile "Ravenscar" at test2.adc:2
t.adb:26:07: violation of restriction "No_Task_Hierarchy"
t.adb:27:07: from profile "Ravenscar" at test2.adc:2
t.adb:27:07: violation of restriction "No_Task_Hierarchy"
t.adb:28:07: from profile "Ravenscar" at test2.adc:2
t.adb:28:07: violation of restriction "No_Task_Hierarchy"
gnatprove: error during generation of Global contracts
t.adb:26:12: tasking in SPARK requires Ravenscar profile (SPARK RM 9(2))
t.adb:26:12: violation of pragma SPARK_Mode at test3.adc:1
t.adb:27:12: tasking in SPARK requires Ravenscar profile (SPARK RM 9(2))
t.adb:27:12: violation of pragma SPARK_Mode at test3.adc:1
t.adb:28:12: tasking in SPARK requires Ravenscar profile (SPARK RM 9(2))
t.adb:28:12: violation of pragma SPARK_Mode at test3.adc:1
t.adb:30:07: tasking in SPARK requires Ravenscar profile (SPARK RM 9(2))
t.adb:30:07: violation of pragma SPARK_Mode at test3.adc:1
t.adb:39:07: tasking in SPARK requires Ravenscar profile (SPARK RM 9(2))
t.adb:39:07: violation of pragma SPARK_Mode at test3.adc:1
t.adb:48:07: tasking in SPARK requires Ravenscar profile (SPARK RM 9(2))
t.adb:48:07: violation of pragma SPARK_Mode at test3.adc:1
gnatprove: error during flow analysis and proof
po_t.ads:14:14: aspect "Part_Of" must apply to package instantiation, object, single protected type or single task type
gnatprove: error during generation of Global contracts
po_t2.ads:14:14: aspect "Part_Of" must apply to package instantiation, object, single protected type or single task type
gnatprove: error during generation of Global contracts
po_t3.adb:1:14: body of package "PO_T3" has unused hidden states
po_t3.adb:1:14: variable "P_Int" defined at line 42
po_t3.adb:2:35: "HIDDEN" is constituent of single protected type "P_Int"
po_t3.adb:2:35: reference to variable "Hidden" cannot appear in this context
gnatprove: error during generation of Global contracts
po_t4.ads:19:07: component "PO" of non-synchronized type "MIX" cannot be synchronized
gnatprove: error during generation of Global contracts
po_t5.adb:2:35: constituent of synchronized state "State" must be synchronized
gnatprove: error during generation of Global contracts
main.adb:8:29: "The_Protected_Int" is constituent of single protected type "P_Int"
main.adb:8:29: reference to variable "The_Protected_Int" cannot appear in this context
main.adb:11:04: "The_Protected_Int" is constituent of single protected type "P_Int"
main.adb:11:04: reference to variable "The_Protected_Int" cannot appear in this context
gnatprove: error during generation of Global contracts
po_t7.adb:11:34: info: overflow check proved (CVC4: 1 VC)
po_t7.adb:18:52: info: overflow check proved (CVC4: 1 VC)
po_t7.ads:11:21: high: variable "Not_CAE" cannot be referenced in precondition of protected operation "Get" (SPARK RM 9(10))
po_t7.ads:14:21: high: variable "Not_CAE" cannot be referenced in precondition of protected operation "Increase" (SPARK RM 9(10))
use_po_t8.adb:3:11: high: more tasks than allowed might queue on protected entry "po_t8.p_int.increase" (restriction No_Entry_Queue active) (SPARK RM 9(11))
use_po_t8.adb:3:11: high: task "po_t8.t" is queueing
use_po_t8.adb:3:11: high: task "use_po_t8" is queueing
use_po_t8.adb:3:11: info: ceiling priority protocol is respected (CVC4: 1 VC)
